kolibri/core/assets/src/views/tab-link/index.vue

```diff
@@ -0,0 +1,109 @@
+<template>
+
+  <li>
+    <router-link :to="link" class="tab" :class="{ 'tab-has-icon-and-title': type === 'icon-and-title' }">
+
+      <div v-if="type === 'icon' || type === 'icon-and-title'" class="tab-icon">
+        <ui-icon :icon="icon" ariaLabel="title" class="icon"/>
+      </div>
+
+      <div v-if="type === 'title' || type === 'icon-and-title'" class="tab-title">
+        {{ title }}
+      </div>
+
+    </router-link>
+  </li>
+
+</template>
+
+
+<script>
+
+  const ValidateLinkObject = require('kolibri.utils.validateLinkObject');
+
+  module.exports = {
+    props: {
+      type: {
+        type: String,
+        validator(type) {
+          return ['title', 'icon', 'icon-and-title'].includes(type);
+        },
+        required: true,
+      },
```

Comment:
I usually think of prop validation as protecting developers against doing things that seem reasonable but are technically incorrect.

Not passing in any title or icon isn't really a reasonable use of the component. To me it's analogous to making an empty `<div/>` and when it doesn't do anything, it's not surprising.

"no title AND no icon" seems like a degenerate case to me which doesn't really need validation. However this is all just intuition - I bet someone could persuade me otherwise :)